agent/netconfig.go

```diff
@@ -22,12 +22,24 @@ import (
 	"net"
 )
 
+const (
+	fullIsolationOption = "full_isolation"
+)
+
+// NetworkRequest specifies messages sent to the
+// agent containing information on how to configure network
+// on its host.
+type NetworkRequest struct {
```

Comment:
I think it's intended to be more generic than k8s endpoint... or you mean any endpoint?